docs/templating-language.md

```diff
@@ -1838,13 +1843,153 @@ the number of instances desired, a unique but consistent identifier associated w
+
+```golang
+{{ nomadVarList "<PREFIX>" }}
+```
+
+Example:
+
+```golang
+{{ range nomadVarList }}
+  {{ . }}
+{{ end }}
+```
+
+You can provide a prefix to filter the path list as an optional parameter to the
+`nomadVarList` function.
+
+```golang
+{{ range nomadVarList "path/to/filter"}}
+  {{ . }}
+{{ end }}
+```
+
+### `nomadVarListSafe`
+
+Same as [`nomadVarList`](#nomadvarlist), but refuses to render template if the
+secure variable list query returns blank/empty data.
+
+#### The `NomadSVMeta` Type
+
+```golang
+type NomadSVMeta struct {
+	Namespace, Path          string
+	CreateIndex, ModifyIndex uint64
+	CreateTime, ModifyTime   nanoTime
+}
+```
+
+The `nanoTime` type contains Unix nanoseconds since the epoch. It's String method
+prints it out as a formatted date/time value. It also has a `Time` method that
+can be used to retrieve a Go `time.Time` for further manipuation.
+
+### `nomadVar`
+
+Query [Nomad][nomad] for the Items at the given secure variable path. If the path
+does not exist or the caller does not have access to it, Consul Template will
+block rendering until the path is available. To avoid blocking, wrap `nomadVar`
+calls with [`nomadVarExists`](#nomadvarexists).
+
+The `nomadVar` function returns a map of NomadSVItems structs. Each member of the
+map corresponds to a member of the secure variable's `Items` collection. Each map
+value also contains helper methods to get the secure variable metadata and a
+link to the parent secure variable.
+
+```golang
+{{ nomadVar "<PATH>" }}
+```
+
+For example:
+
+Supposing a secure variable exists at the path `nomad/jobs/redis`, with a single
+key/value pair in its Items collection—`maxconns`:`15`
+
+```golang
+{{ with nomadVar "nomad/jobs/redis" }}{{ .maxconns }}{{ end }}
+```
+
+renders
+
+```text
+15
+```
+
+#### The `NomadSVItems` Type
+
+Calls to the `nomadVar` function return a value of type `NomadSVItems`. This
+value is a map of `string` to `NomadSVItem` elements. It provides some
+convenience methods:
+
+* `Keys`: produces a sorted list of keys to this `NomadSVItems` map.
+
+* `Values`: produces a key-sorted list.
+
+* `Tuples`: produces a key-sorted list of K,V tuple structs.
+
+* `Metadata`: returns this collection's parent metadata as a `NomadSVMeta`
+
+* `Parent`: returns the consul-template version (NomadSecureVariable) of a full
+    secure variable, which has the Metadata values and Items collection as peers.
+
+#### The `NomadSVItem` Type
+
+Specific item of a NomadSVItems collection that are directly accessed by name or
+the elements that are obtained while ranging the return of `nomadVar` are of type
+`NomadSVItem`.
+
+```golang
+type NomadSVItem struct {
+	Key, Value string
+}
+```
+
+`NomadSVItem` objects also provide helper methods:
+
+* `Metadata`: returns this item's parent's metadata as a `NomadSVMeta`
+
+* `Parent`: returns the consul-template version (NomadSecureVariable) of the
+    secure variable that contains this item.
```

Comment:
I suspect helper methods are not-too-unintuitive for go devs but for operators it might be useful to specifically document an example of their use here.